File: backend/internal/model/dto/notification.go
package dto

import (
	"time"

	"github.com/google/uuid"
	"github.com/guregu/null/v6"

	"github.com/lesprivate/backend/internal/model"
)

type GetNotificationsRequest struct {
	IsDismissed null.Bool `form:"isDismissed"`
	IsRead      null.Bool `form:"isRead"`
	IsDeleted   null.Bool `form:"isDeleted"`
	model.Pagination
}

type Notification struct {
	ID           uuid.UUID              `json:"id"`
	Title        string                 `json:"title"`
	Message      string                 `json:"message"`
	Type         model.NotificationType `json:"type"`
	Link         string                 `json:"link"`
	IsRead       bool                   `json:"isRead"`
	IsDismissed  bool                   `json:"isDismissed"`
	IsDeleteable bool                   `json:"isDeletable"`
	CreatedAt    time.Time              `json:"createdAt"`
}

File: mentor/utils/types/notification.ts
export interface Notification {
    id: string;
    title: string;
    message: string;
    link?: string;
    type: "info" | "success" | "warning" | "error";
    is_read: boolean;
    created_at: string;
}

export interface NotificationFilters {
    page?: number;
    limit?: number;
    isRead?: boolean;
}
